ReactAndroid/src/main/java/com/facebook/react/modules/intent/IntentModule.java

```diff
@@ -34,24 +34,57 @@ public String getName() {
   }
 
   /**
+   * Return the URL the activity was started with
+   *
+   * @param callback a callback which is called with the initial URL
+   */
+  @ReactMethod
+  public void getInitialURL(Callback callback) {
+    try {
+      Activity currentActivity = getCurrentActivity();
+      String initialURL = null;
+
+      if (currentActivity != null) {
+        Intent intent = currentActivity.getIntent();
+        String action = intent.getAction();
+        Uri uri = intent.getData();
+
+        if (Intent.ACTION_VIEW.equals(action) && uri != null) {
+          initialURL = uri.toString();
+        }
+      }
+
+      callback.invoke(initialURL);
+    } catch (Exception e) {
+      throw new JSApplicationIllegalArgumentException(
```

Comment:
Can you add an error callback (all `@ReactMethod`s have two callbacks) and call it here instead of throwing an exception? You should probably still log the full exception details, though, to make it easier to debug problems.